libraries/Microsoft.Bot.PublishValidation/README.md

```diff
@@ -0,0 +1,77 @@
+# Microsoft Bot Framework Publish Validation for .NET
+
+This tool is used for validating a Bot's Configuration prior to publishing it, in order to prevent common issues like missing endpoints.
+
+## Target Framework
+
+* .NET Core 2.1
+
+## How to Install
+  
+  ```
+  PM> Install-Package Microsoft.Bot.PublishValidation
+  ```
+  
+## How to use
+
+There are several validation checks available at the moment. These are the following:
+- **ForbidSpacesInProjectName**: 
+    - Possible values: 'True' or 'False'
+    - Description: No spaces in the Project's name (this is due to a [bug in .NET/Azure](https://github.com/aspnet/websdk/issues/237))
+- **RequireBotFile**:
+    - Possible values: 'True' or 'False'
+    - Description: Existence of a '.bot' file
+- **RequireEndpoints**: 
+    - Possible values: List of comma separated values
+    - Description: Required endpoints inside the configuration file
+- **ForbidEndpoints**: 
+    - Possible values: List of comma separated values
+    - Description: Forbidden endpoints inside the configuration file
+- **RequireLuisKey**: 
+    - Possible values: 'True' or 'False'
+    - Description: Existence of LUIS key inside the configuration file
+- **RequireQnAMakerKey**: 
+    - Possible values: 'True' or 'False'
+    - Description: Existence of QnA Maker key inside the configuration file
+
+By default, these validations are all set like this:
+
+> ForbidSpacesInProjectName: True
+>
+> RequireBotFile: True
+>
+> RequireEndpoints: Production
+>
+> ForbidEndpoints: Development
+>
+> RequireLuisKey: True
+>
+> RequireQnAMakerKey: True
+
+In order to change the behavior of this validations, you can create your own properties in your `.csproj` file by adding the following code:
+
+```.csproj
+<PropertyGroup>
+    <ForbidSpacesInProjectName>True</ForbidSpacesInProjectName>
```

Comment:
I don't understand how these XML methods are mapped to the configuration class. I'm clearly missing something obvious.

Reply:
In the [Microsoft.Bot.PublishValidation.targets] (/libraries/Microsoft.Bot.PublishValidation/build/Microsoft.Bot.PublishValidation.targets) there are some properties that take the values of these properties located in the `.csproj` file. We added some comments in the `.targets` file for better understanding how it works.

Reply:
In the [Microsoft.Bot.PublishValidation.targets] (/libraries/Microsoft.Bot.PublishValidation/build/Microsoft.Bot.PublishValidation.targets) there are some properties that take the values of these properties located in the `.csproj` file. We added some comments in the `.targets` file for better understanding how it works.